### DISCUSSION TITLE
X-Livewire header and other empty headers are being removed when used in conjunction with a Vapor Load Balancer

### DISCUSSION THREAD

**Discussion by duprayce10r8mn** [livewire/livewire#8333] [7. Bugs (archived)]
### Livewire version

v3.4.10

### Laravel version

v11.4.0

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

1. Spin up a new Laravel project.
2. Install Livewire.
3. Install Laravel Vapor.
4. Switch to using a Vapor Load Balancer instead of Default API Gateway.
5. Make a new Livewire component and add listeners for a dispatch event.
6. Send Livewire request by using a `$dispatch` call.
7. Look for `X-Livewire` header in the request, it will be:
a. Present in Local testing.
b. Present when using Laravel Vapor with Default API Gateway.
c. NOT present when Load Balancer is used.

### Repository or Wirebox URL

https://github.com/duprayce10r8mn/laravel-livewire-demo

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by ErikDohmen** [comment]
@duprayce10r8mn 
We've noticed the same with our loadbalancer as being a bug, but not fixed (yet): https://my.f5.com/s/article/K000139285
I've found a solution as per the comment of Caleb here (https://github.com/livewire/livewire/pull/7561#issuecomment-1873422200)
We fixed this by setting the header to true
```
Livewire.hook('request', ({ options }) => {
    options.headers['X-Livewire'] = true;
})
```

**Reply by mateusgalasso** [reply]
Where did you add this? In some middleware?

**Reply by mikwiat** [reply]
@mateusgalasso 

Try to add it in one of your js-files:

`document.addEventListener('livewire:initialized', () => {
    Livewire.hook('request', ({ options }) => {
        options.headers['X-Livewire'] = true;
    });
});`

It should work!

**Reply by mateusgalasso** [reply]
It worked and saved my life, thank you.
I love this community ❤️

**Reply by yassir3wad** [reply]
> @duprayce10r8mn We've noticed the same with our loadbalancer as being a bug, but not fixed (yet): https://my.f5.com/s/article/K000139285 I've found a solution as per the comment of Caleb here ([#7561 (comment)](https://github.com/livewire/livewire/pull/7561#issuecomment-1873422200)) We fixed this by setting the header to true
> 
> ```
> Livewire.hook('request', ({ options }) => {
>     options.headers['X-Livewire'] = true;
> })
> ```

This fixed the issue. Thanks
